h2o-k8s/tests/clustering/docker-init-script.sh

```diff
@@ -9,21 +9,49 @@ fi
 cd $H2O_BASE/h2o-k8s/tests/clustering/
 k3d --version
 k3d delete
-k3d create -v $H2O_BASE/build/h2o.jar:$H2O_BASE/build/h2o.jar --registries-file registries.yaml --publish 8080:80 --api-port localhost:6444 --server-arg --tls-san="127.0.0.1" --wait 120 
+k3d create -v "$H2O_BASE":"$H2O_BASE" --registries-file registries.yaml --publish 8080:80 --api-port localhost:6444 --server-arg --tls-san="127.0.0.1" --wait 120
 export KUBECONFIG="$(k3d get-kubeconfig --name='k3s-default')"
 kubectl cluster-info
 sleep 15 # Making sure the default namespace is initialized. The --wait flag does not guarantee this.
 kubectl get namespaces
+# Deploy H2O-3 Cluster as defined by Helm template in h2o-helm subproject
+# Also tests correctness of the H2O HELM chart
 envsubst < testvalues-template.yaml >> testvalues.yaml
 helm install -f testvalues.yaml h2o $H2O_BASE/h2o-helm --kubeconfig $KUBECONFIG --dry-run # Shows resulting YAML
 helm install -f testvalues.yaml h2o $H2O_BASE/h2o-helm --kubeconfig $KUBECONFIG
+# Use helm built-in test pod
 helm test h2o
+# After the deployment, show status of H2O-related K8S resources
 kubectl logs h2o-h2o-3-test-connection
 kubectl get ingresses
 kubectl describe pods
+# Check H2O is clustered with a script independent from the one in h2o-helm triggered previously with 'helm test'
 timeout 120s bash h2o-cluster-check.sh
-export EXIT_STATUS=$?
+export CLUSTER_EXIT_CODE=$?
 kubectl get pods
 kubectl get nodes
+# To save resources in Jenkins pipeline (e.g. CPUs are limited), remove the H2O cluster deployed via HELM
+helm uninstall h2o 
+# Test assisted clustering regime by adding h2o-clustering artifact into the H2O container using volumes.
+envsubst < h2o-assisted-template.yaml >> h2o-assisted.yaml
```

Comment:
Does this mean it is all tested in a single stage?

Would it be better to have different stages? If one thing doesn't work - do the other tests run?

Reply:
If one deployment check fails, the exit value is stored and the script continues. Read line 56:

```bash
export EXIT_STATUS=$(if [ "$CLUSTER_EXIT_CODE" -eq 1 ] || [ "$ASSISTED_EXIT_CODE" -eq 1 ]; then echo 1; else echo 0; fi)
exit $EXIT_STATUS
```

This stage's exit code is **decided at the end**, if one of the clustering stages fails. This means both stages are ran. The `kubectl get *` diagnostics in the middle provide a detailed look into the cluster (especially the `kubectl describe *` commands), allowing for detailed overview of the cluster state in case of test failure.

Having separate stages for different ways of clustering would be too much. Also, this test belongs into this suite, as it tests a specific use case - Kubernetes clustering using a generic assisted clustering H2OEmbeddeConfigProvider.